src/ray/common/task/task_spec.cc

```diff
@@ -240,6 +245,25 @@ const ResourceSet &TaskSpecification::GetRequiredResources() const {
   return *required_resources_;
 }
 
+const rpc::SchedulingStrategy &TaskSpecification::GetSchedulingStrategy() const {
+  return message_->scheduling_strategy();
+}
+
+bool TaskSpecification::IsNodeSchedulingStrategy() const {
+  return GetSchedulingStrategy().scheduling_strategy_case() ==
+         rpc::SchedulingStrategy::SchedulingStrategyCase::kNodeSchedulingStrategy;
+}
+
+NodeID TaskSpecification::GetNodeSchedulingStrategyNodeId() const {
+  RAY_CHECK(IsNodeSchedulingStrategy());
```

Comment:
Could we return std::optional<NodeID> and check at a higher level? It's weird that an accessor can CHECK fail.

Reply:
Either case the caller needs to do some check: either check before calling or check after calling. All the other accessors in TaskSpecification are using the RAY_CHECK style as well instead of returning std::optional. I'm fine with either, if you feel strongly about it, I can change it.